### ISSUE TITLE
False positive E0308 on boxed slice.

### ISSUE THREAD

**Issue by harrybraviner** [intellij-rust/intellij-rust#6484]
## Environment

* **IntelliJ Rust plugin version:** 4.76
* **Rust toolchain version:** 1.48.0
* **IDE name and version:** CLion 2020.3
* **Operating system:** Linux

## Problem description

When a function takes an argument of type `Box<[f64]>`, the IDE flags passing such a type in as an E0308. But the code will compile. Conversely, applying the recommended fix leads to code that does not compile.

Note that I'm quite new to Rust, so I'm not sure that what I'm doing is valid rust (I don't think my function argument is either a valid array or array slice). But the errors message produced by the IDE is still quite misleading and unhelpful.

## Steps to reproduce

The following code results in the plugin flagging E0308 on the call to `foo`:
```rust
fn foo(x: Box<[f64]>) {
    panic!("Not implemented");
}

fn main() {
    foo(Box::new([0f64]));
}
```
However, this does compile.

The recommended fix changes the argument of the call to `foo`:
```rust
fn foo(x: Box<[f64]>) {
    panic!("Not implemented");
}

fn main() {
    foo(Box::new(*[0f64]));
}
```
But this *does* give compilation errors:
```
error[E0614]: type `[f64; 1]` cannot be dereferenced
 --> src/main.rs:6:18
  |
6 |     foo(Box::new(*[0f64]));
  |                  ^^^^^^^

error[E0277]: the size for values of type `[f64]` cannot be known at compilation time
 --> src/main.rs:6:9
  |
6 |     foo(Box::new(*[0f64]));
  |         ^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
  |
  = help: the trait `Sized` is not implemented for `[f64]`
  = note: required by `Box::<T>::new`

error[E0277]: the size for values of type `[f64]` cannot be known at compilation time
 --> src/main.rs:6:9
  |
6 |     foo(Box::new(*[0f64]));
  |         ^^^^^^^^ doesn't have a size known at compile-time
  |
  = help: the trait `Sized` is not implemented for `[f64]`
  = note: all function arguments must have a statically known size
```

**Comment by neonaot** [comment]
Sorry that there was no answer for a long time. 
I cannot reproduce this false positive bug in Clion 2021.2 and 2021.1, probably it is already fixed 😄 .
